### PR TITLE
Adding TapToPlace

### LINKED ISSUE #17: Consider adding TapToPlaceParent like functionality
**Scenario:**

Adam is a developer who wants an easy way to move his objects around. Place them where they matter to him.
He wants a quick way to tap on an object, gaze at a different location and then tap again to place the object at that location.

**Proposed Solution:**

Add a new script called TapToPlaceParent or something similar to either the Input or Utilities folder.

### PR DESCRIPTION
**Pull request by jochao**
Adding TapToPlace script which helps place an object on a real surface.

Closes #17

### CODE REVIEW

File: Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    /// Put this script on the object you want to be able to move. 
+    /// Users will be able to select objects, gaze elsewhere, and perform the
+    /// select gesture again to place.
+    /// This script is used in conjunction with GazeManager, GestureManager,
+    /// and SpatialMappingManager.
+    /// </summary>
+
+    public partial class TapToPlace : MonoBehaviour
+    {
+
```
Comment (keveleigh) [inline Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs]: Please remove this space

File: Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        // Called by GazeGestureManager when the user performs a Select gesture.
+        void OnSelect()
+        {
+            // On each Select gesture, toggle whether the user is in placing mode.
+            placing = !placing;
+
+            // If the user is in placing mode, display the spatial mapping mesh.
+            if (placing)
+            {
+
```
Comment (keveleigh) [inline Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs]: This space too

File: README.md [hunk 3]
```diff
@@ -363,6 +363,10 @@ Generates planes and classifies them by type (wall, ceiling, floor, table, unkno
 
 **UnknownMaterial** Material to use when rendering unknown plane types.
 
+#### TapToPlace.cs
+Simple script to add to a gameobject that allows users to select and place the gameobject along the spatial mapping mesh.
```
Comment (NeerajW) [inline README.md]: Consider adding a comment for the Test you added

File: Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// The TapToPlace class is a basic way to enable users to move objects 
+    /// and place them on real world surfaces.
+    /// Put this script on the object you want to be able to move. 
+    /// Users will be able to select objects, gaze elsewhere, and perform the
+    /// select gesture again to place.
```
Comment (NeerajW) [inline Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs]: Tap gesture

File: Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    /// This script is used in conjunction with GazeManager, GestureManager,
+    /// and SpatialMappingManager.
+    /// </summary>
+
+    public partial class TapToPlace : MonoBehaviour
+    {
+        bool placing = false;
+
+        // Called by GazeGestureManager when the user performs a Select gesture.
```
Comment (NeerajW) [inline Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs]: ... performs a Tap gesture.

File: Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        void OnSelect()
+        {
+            // On each Select gesture, toggle whether the user is in placing mode.
+            placing = !placing;
+
+            // If the user is in placing mode, display the spatial mapping mesh.
+            if (placing)
+            {
+                SpatialMappingManager.Instance.DrawVisualMeshes = true;
```
Comment (NeerajW) [inline Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlace.cs]: What if SpatialMappingManager is null?

**Pull request merged**